File: backend/src/main/java/com/app/backend/responses/thongke/AdminSummaryStatsResponse.java
package com.app.backend.responses.thongke;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AdminSummaryStatsResponse {

    // ========== NGƯỜI DÙNG ==========
    @JsonProperty("tong_nguoi_dung")
    private long tongNguoiDung;
    
    @JsonProperty("nguoi_dung_active")
    private long nguoiDungActive;
    
    @JsonProperty("nguoi_dung_blocked")
    private long nguoiDungBlocked;
    
    @JsonProperty("nguoi_dung_deleted")
    private long nguoiDungDeleted;
    
    @JsonProperty("so_admin")
    private long soAdmin;
    
    @JsonProperty("nguoi_dung_moi_hom_nay")
    private long nguoiDungMoiHomNay;

    // ========== TRẬN ĐẤU ==========
    @JsonProperty("tong_tran_dau")
    private long tongTranDau;
    
    @JsonProperty("tran_dang_cho")
    private long tranDangCho;
    
    @JsonProperty("tran_dang_dien_ra")
    private long tranDangDienRa;
    
    @JsonProperty("tran_da_ket_thuc")
    private long tranDaKetThuc;
    
    @JsonProperty("tran_da_huy")
    private long tranDaHuy;
    
    @JsonProperty("tran_hom_nay")
    private long tranHomNay;

    // ========== BỘ CÂU HỎI / CÂU HỎI ==========
    @JsonProperty("tong_bo_cau_hoi")
    private long tongBoCauHoi;
    
    @JsonProperty("bo_cau_hoi_da_duyet")
    private long boCauHoiDaDuyet;
    
    @JsonProperty("bo_cau_hoi_cho_duyet")
    private long boCauHoiChoDuyet;
    
    @JsonProperty("bo_cau_hoi_tu_choi")
    private long boCauHoiTuChoi;
    
    @JsonProperty("tong_cau_hoi")
    private long tongCauHoi;

    // ========== KHÓA HỌC ==========
    @JsonProperty("tong_khoa_hoc")
    private long tongKhoaHoc;
    
    @JsonProperty("khoa_hoc_published")
    private long khoaHocPublished;
    
    @JsonProperty("khoa_hoc_draft")
    private long khoaHocDraft;
}

File: frontend/src/app/responses/thongke/admin-summary-stats-response.ts
// src/app/responses/thongke/admin-summary-stats-response.ts

export interface AdminSummaryStatsResponse {
  // Người dùng
  tong_nguoi_dung: number;
  nguoi_dung_active: number;
  nguoi_dung_blocked: number;
  nguoi_dung_deleted: number;
  so_admin: number;
  nguoi_dung_moi_hom_nay: number;

  // Trận đấu
  tong_tran_dau: number;
  tran_dang_cho: number;
  tran_dang_dien_ra: number;
  tran_da_ket_thuc: number;
  tran_da_huy: number;
  tran_hom_nay: number;

  // Bộ câu hỏi
  tong_bo_cau_hoi: number;
  bo_cau_hoi_da_duyet: number;
  bo_cau_hoi_cho_duyet: number;
  bo_cau_hoi_tu_choi: number;
  tong_cau_hoi: number;

  // Khóa học
  tong_khoa_hoc: number;
  khoa_hoc_published: number;
  khoa_hoc_draft: number;
}
